### PR TITLE
Sprint4/add fields to profile

### PR DESCRIPTION
**Pull request by christinajiyunlee**
DB Changes:
- Changes were completed last sprint but not reviewed in detail, as it was "pre-work"
- README should be inspected to ensure DB makes sense closing this PR.

Added the following fields to profile view:
- application status
- address
- date of birth
- current occupation
- canadian id provided (boolean)
- health insurance provided (boolean)
- government issued id provided (boolean)

Logic added:
- When application status is "Began Application" (default and means app saved but not submitted), a blue dialogue appears reminding user to submit their app for employers to see. Once app is submitted, this dialogue disappears.
- If some non-mandatory fields have Null values in DB (Address, DOB, current occupation), field will display "Not Provided". If they have real values in DB, real values will be displayed. 

The edits to the view profile is reflected on the admin's "approve application" page. This is viewable by clicking on a profile from the search results page.

### CODE REVIEW

File: server/searchPage.js [hunk 1]
```diff
@@ -10,9 +10,12 @@ const pool = new Pool({
 
 
 module.exports.applySearch = async event => {
-		// const selectAllCleaners = "SELECT * FROM cleaner_profile"
+		/* @Gurleen: This is a SELECT statement to help with the filter story
```
Comment (nicolasabousawan) [inline server/searchPage.js:13]: Thanks for doing this, this will help later tickets greatly - lets merge this PR ASAP so its deployed to the team

File: ui/src/Components/ProfilePage.jsx [hunk 2]
```diff
@@ -112,39 +112,78 @@ export default class ProfilePage extends React.Component {
...
           ) : (
-            <div className="row-buffer side-buffer left-align">
-              <h4>{this.state.profile.result.first_name} {' '}
-              {this.state.profile.result.last_name}</h4>
-              {this.state.profile.result.city ? <H4>Servicing the {this.state.profile.result.city} area</H4> : <div/>}
-              <Table className="row-buffer" responsive>
+            <div className="row-buffer side-buffer">
+               {this.state.profile.result.app_status == 'Began Application' ? (
+                  <Alert className="width-fit" color="primary">
+                    Your application is in draft mode.
```
Comment (nicolasabousawan) [inline ui/src/Components/ProfilePage.jsx:154]: Thoughts on making this a global variable - usually const are much easily manipulated at the top of component set up rather in-line... thoughts?

Comment (nicolasabousawan) [reply ui/src/Components/ProfilePage.jsx:154]: This is not a blocking concern, but more maintainability

**Comment by nicolasabousawan** [comment]
Validated Alert in Edit Profile
image

**Comment by nicolasabousawan** [comment]
Validated Field Validation **(NOTE: NOT PART OF THIS TICKET)** 
image

**Comment by nicolasabousawan** [comment]
@christinajiyunlee Aren't references supposed to be displayed as well? I do not see them in the code?

**Comment by nicolasabousawan** [comment]
Tool Pic URL included
image

**Comment by nicolasabousawan** [comment]
Admin view validated as well
image

**Comment by nicolasabousawan** [comment]
AMAZING! 
Only 1 open question about the references (that is approval blocking) and then a copy suggestion (that is low priority) but if you could address the referrals, then I am good to approve.

File: server/profile.js [hunk 3]
```diff
@@ -42,6 +42,13 @@ module.exports.getProfile = async (event) => {
     }
   }
 
+  const selectRefs = "SELECT * FROM has_ref LEFT JOIN reference ON has_ref.ref_id = reference.ref_id WHERE has_ref.profile_id = $1";
+  const values2 = [profileId]
+
+  const resultRefs = await pool.query(selectRefs, values2);
+  console.log(resultRefs)
```
Comment (nicolasabousawan) [inline server/profile.js:49]: Do we still need to log?

File: ui/src/Components/ProfilePage.jsx (hunk 2 above)
Comment (nicolasabousawan) [reply ui/src/Components/ProfilePage.jsx:118]: Was this comment looked at?

**Comment by nicolasabousawan** [comment]
Prematurely opened a review - sorry 😬

File: server/profile.js [hunk 4]
```diff
@@ -6,6 +6,12 @@ const pool = new Pool({
     database: process.env.database,
     password: process.env.password,
     port: process.env.port,
+    // user: 'master',
+    // host: 'aoc-db-dev.cjynw6x3q1ly.us-west-1.rds.amazonaws.com',
+    // database: 'aocdb',
```
Comment (kvptkr) [inline server/profile.js]: lmao at least I wasn't the one to leak things this time!!!!! We're gonna have to rotate the db creds lol

File: server/profile.js [hunk 5]
```diff
@@ -42,6 +48,13 @@ module.exports.getProfile = async (event) => {
     }
   }
 
+  const selectRefs = "SELECT * FROM has_ref LEFT JOIN reference ON has_ref.ref_id = reference.ref_id WHERE has_ref.profile_id = $1";
+  const values2 = [profileId]
```
Comment (kvptkr) [inline server/profile.js:46]: why is the profileId in square brackets?

File: server/profile.js [hunk 6]
```diff
@@ -42,6 +48,13 @@ module.exports.getProfile = async (event) => {
...
+
+  const resultRefs = await pool.query(selectRefs, values2);
+  // console.log(resultRefs)
```
Comment (kvptkr) [inline server/profile.js]: delete console.log call

File: ui/src/Components/ProfilePage.jsx [hunk 7]
```diff
@@ -94,13 +94,34 @@ export default class ProfilePage extends React.Component {
...
+    // Builds table with JSON object items passed in
+
+    if (refLen == 0) {
+      references.push(<tr><th>Reference</th><td>Not Provided</td></tr>)
+    }
+
+    for (var i = 0; i < refLen; i++) {
+      references.push(<tr><th>Reference {i+1}</th>
+        <td><div>{refs[i].first_name} {refs[i].last_name}</div>
+        <div>{refs[i].email}</div><div>{refs[i].contact_num.substring(0,3)}-{refs[i].contact_num.substring(3,6)}-{refs[i].contact_num.substring(6,10)} x {refs[i].contact_ext}</div>
```
Comment (kvptkr) [inline ui/src/Components/ProfilePage.jsx]: can we call a method somewhere to format this for us? Was a little confused at to what this actually meant.

File: ui/src/Components/ProfilePage.jsx [hunk 8]
```diff
@@ -112,39 +133,83 @@ export default class ProfilePage extends React.Component {
...
-                    <td>Email</td>
+                    <th>Email</th>
                     <td>{this.state.profile.result.email}</td>
                   </tr>
                   <tr>
-                    <td>Phone Number</td>
-                    <td>{this.state.profile.result.contact_num}</td>
+                    <th>Phone Number</th>
+                    {this.state.profile.result.contact_ext ? (
+                    <td>{this.state.profile.result.contact_num.substring(0,3)}-{this.state.profile.result.contact_num.substring(3,6)}-{this.state.profile.result.contact_num.substring(6,10)}  x  {this.state.profile.result.contact_ext}</td>
```
Comment (kvptkr) [inline ui/src/Components/ProfilePage.jsx]: Yeah, the fact that we do the same thing more than once really makes me think that we should create a method to format the phone numbers properly.

File: ui/src/Components/ProfilePage.jsx [hunk 9]
```diff
@@ -112,39 +133,83 @@ export default class ProfilePage extends React.Component {
...
+                    {this.state.profile.result.address ? (
+                        <td><div>{this.state.profile.result.address}</div>
+                        <div>{this.state.profile.result.city},  {this.state.profile.result.postal_code.substring(0,3)} {this.state.profile.result.postal_code.substring(3,6)}   {this.state.profile.result.province}</div>
+                        </td>
+                      ) : <td>Not Provided</td>}
+                  </tr>
+                  <tr>
+                    <th>Date of Birth (MM/DD/YYYY)</th>
+                    {this.state.profile.result.dob ? (
+                      <td>{this.state.profile.result.dob.substring(0,2)}/{this.state.profile.result.dob.substring(2,4)}/{this.state.profile.result.dob.substring(4,8)}</td>
```
Comment (kvptkr) [inline ui/src/Components/ProfilePage.jsx:189]: Just a thought but given the number of times we call `this.state.profile.result` we should probs extract it out to a variable. We're only fetching and setting this.state.profile once, so I think that should be fine.

File: ui/src/Components/App.css [hunk 10]
```diff
@@ -100,4 +111,20 @@ p {
...
+
+.push-left {
+  float: left;
+  display: table;
+}
+
+.push-right {
+  float: right;
+  display: table;
+
```
Comment (kvptkr) [inline ui/src/Components/App.css]: extra line

File: ui/src/Components/ProfilePage.jsx [hunk 11]
```diff
@@ -94,13 +94,34 @@ export default class ProfilePage extends React.Component {
...
           </Form>
 
         )
       }
   }
+
+
+  if (this.state.profile && this.state.profile.resultRefs) {
+    const refs = this.state.profile.resultRefs;
+    var refLen = Object.keys(refs).length;
```
Comment (kvptkr) [inline ui/src/Components/ProfilePage.jsx]: Little nit: the concept of `vars` I think are now deprecate. I think you're supposed to use `let` in their place now.

File: ui/src/Components/ProfilePage.jsx [hunk 12]
```diff
@@ -112,39 +133,83 @@ export default class ProfilePage extends React.Component {
...
+                  <tr>
+                    <th>Canadian ID Provided?</th>
+                    {this.state.profile.result.canadian_id ? (
+                    <td>Yes</td>
+                  ) : <td>No</td>}
                   </tr>
                   <tr>
-                    <td>Extension</td>
-                    <td>{this.state.profile.result.contact_ext}</td>
+                    <th>Health Insurance Provided?</th>
```
Comment (kvptkr) [inline ui/src/Components/ProfilePage.jsx:208]: Great catch changing up the `<td>` -> `<th>`

**Review by kvptkr — CHANGES_REQUESTED** [review]
Validated locally that the changes were showing up and that there were no breaking changes.

@christinajiyunlee make sure that you deploy this to heroku and make sure your changes work there!

Looks really good Christina, I left some requests for some small changes + questions. I think this really is phenomenal work considering you just started learning react!!

File: server/profile.js (hunk 4 above)
Comment (christinajiyunlee) [reply server/profile.js]: 😥😥

File: server/profile.js (hunk 5 above)
Comment (christinajiyunlee) [reply server/profile.js:46]: I followed the format of the other SELECT statement, but I believe because it should be provided as an array (this is me inferring though)

File: server/profile.js (hunk 6 above)
Comment (christinajiyunlee) [reply server/profile.js]: ✔

File: ui/src/Components/ProfilePage.jsx (hunk 7 above)
Comment (christinajiyunlee) [reply ui/src/Components/ProfilePage.jsx]: I agree it may be hard to follow, especially because it's outside of the rendered page code.
Created a formatAddress() function to resolve this!

File: ui/src/Components/ProfilePage.jsx (hunk 11 above)
Comment (christinajiyunlee) [reply ui/src/Components/ProfilePage.jsx]: ✔

File: ui/src/Components/ProfilePage.jsx (hunk 12 above)
Comment (christinajiyunlee) [reply ui/src/Components/ProfilePage.jsx:208]: 🙆‍♀️

File: ui/src/Components/App.css (hunk 10 above)
Comment (christinajiyunlee) [reply ui/src/Components/App.css]: ✔

File: server/profile.js (hunk 5 above)
Comment (nicolasabousawan) [reply server/profile.js:46]: Not necessarily... JS is really smart at inferring the appropriate data type. I don't mind leaving it in an array jic ion the future the SQL stmt may change. But if its only value, you could have passed it directly into query(), FYI :)

File: ui/src/Components/ProfilePage.jsx (hunk 8 above)
Comment (nicolasabousawan) [reply ui/src/Components/ProfilePage.jsx]: Just to call out, this is expecting all numbers to be US/Can for now... can we add that as a comment that this is our current assumption. No need to change the logic bc I doubt this will be WorldWide anytime soon

**Comment by nicolasabousawan** [comment]
I will leave Kanav to provide final approval here - but I am on board with all the comments in the PR and will leave them to y'all to address and discuss. Great work team :)

File: ui/src/Components/ProfilePage.jsx (hunk 8 above)
Comment (christinajiyunlee) [reply ui/src/Components/ProfilePage.jsx]: Yes exactly @nicolasabousawan ! Especially since our create profile phone number field only accepts 10 digit phone numbers, which is consistent with US/Can but not world wide

Comment (christinajiyunlee) [reply ui/src/Components/ProfilePage.jsx]: @kvptkr Just pushed with formatter functions added :)

File: ui/src/Components/ProfilePage.jsx (hunk 9 above)
Comment (christinajiyunlee) [reply ui/src/Components/ProfilePage.jsx:189]: good point, maybe we'll want to eventually fix this redundancy across all the JSX files, in one sweep

Comment (kvptkr) [reply ui/src/Components/ProfilePage.jsx:189]: that works with me

**Review by kvptkr — APPROVED** [review]
